nomt/src/beatree/ops/update/branch_updater.rs

```diff
@@ -337,63 +493,125 @@ impl BranchUpdater {
             return builder.finish();
         };
 
-        let base_prefix_compressed = base.node.prefix_compressed() as usize;
-        let mut pending_chunk = None;
-        for i in 0..gauge.prefix_compressed_items() {
-            match (&ops[i], &mut pending_chunk) {
-                // start a chunk
-                (BranchOp::Keep(pos, _), None) if *pos < base_prefix_compressed => {
-                    pending_chunk = Some((*pos, pos + 1))
+        // This second phase of joining Update and KeepChunk into a unique update chunk is performed
+        // for two reasons:
+        //
+        // 1. It could often happen that the sequence of KeepChunk are interleaved by Update with only a change
+        // in the node pointers
+        // 2. To avoid keeping all the update information within the BranchOp::KeepChunk because it would require
+        // further allocations
+        let apply_chunk =
```

Comment:
I don't understand the underlying motivation for this piece of code. It looks like it adds a lot of complexity.

Was performance too slow with it? Is it causing a bug? It looks to me like we could just remove this entire piece of logic and save about 100LoC with dense conditionals. Can you elaborate on reason (2) stated here - all the original data for `KeepChunk` is stored within the base branch, so what causes additional allocations? The signature of push_chunk (base, start, end, ops) should work fine with that.

Our happy-path is